### DISCUSSION TITLE
How to model CSP?

### DISCUSSION THREAD

**Discussion by fikrikscg** [NREL/ReEDS-2.0#110] [Q&A: Features] [answered]
Hi,

How did you model operations of concentrated solar power units? I looked at the documentation, but I did not see mathematical modeling.

Since CSP is solar-powered, it should be treated as renewable source. Since it can store, it should be treated as storage units. Hence, I am bit confused how to combine these two aspects.

It would be very helpful if you describe how to model it, or point to the mathematical modeling, if any?

Kind regards

**Comment by wesleyjcole** [comment, marked as answer]
We have CSP with and without storage in the model.  The CSP without storage is treated similar to PV.  The CSP with storage is defined by resource class, solar multiple, and storage duration.  This CSP is dispatchable, but still limited by storage duration in its dispatch.

You can see how the CSP is modeled in c_supply_model.gms.  For example, eq_storage_level shows how the CSP generation interacts with the storage duration.  And you can look for places in the code with "csp_storage" as those elements will apply specifically to CSP with storage.

**Reply by fikrikscg** [reply]
Thank you so much for your answer. I have the following follow-up questions:

1) Do you have capacity factor for CSP? Or is it assumed 100% for all time slices?
2) If you have capacity factor, how do you treat it? Is it inflow parameter for CSP? Similar to inflow for reservoir hydro unit...

I hope my questions are clear.

Kind regards

**Reply by vcelloho** [reply]
1. Capacity factor for CSP  is calculated using the script in input_processing folder called cfgather. This informs the time slice level capacity factor availability for CSP.  The formula is effectively Capacity * Solar Multiple * Capacity Factor for CSP with storage, this will be specific to each plant design, location, and time slice.
2. The treatment varies between CSP with storage and without storage. With storage it is treated as an inflow parameter. https://github.com/NREL/ReEDS_OpenAccess/blob/023c9c197b1c5af77e659fa0d1d1625701aa587f/c_supplymodel.gms#L2201 For CSP without storage it's treated the same as wind and the CF directly informs generation.

**Reply by fikrikscg** [reply]
Thank you. It is very clear now. One more very quick question. What is solar multiple?

From your description, I understand that for a 100 MW CSP with storage, inflow amount is 100 * 0.2 * 0.7 = 14 MW at a time period, with the assumption that 0.2 is the capacity factor at that time period and 0.7 is solar multiple. With your answer, solar multiple will be also be clear for me.

**Reply by vcelloho** [reply]
The solar multiple is a ratio that meausres the solar field size relative to the capacity of the power block. It's analogous in certain ways to inverter loading ratio for PV. The key difference is that for CSP with storage there's an incentive to have a solar multiple greater than 1, as excess energy can be built up in the thermal storage system.

**Reply by fikrikscg** [reply]
Okay, I understand.

Can I see capacity factors somewhere in this repo? Based on online documents, it is supposed to be around 70% annual (for csp with thermal storage). But, I am more interested in how it changes from one time slice to another.
